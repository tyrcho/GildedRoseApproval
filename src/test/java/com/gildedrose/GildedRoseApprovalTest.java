package com.gildedrose;


import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static com.gildedrose.ItemNames.*;


class GildedRoseApprovalTest {

    @Test
    void updateQuality_normal_shouldDecrease() throws Exception {
        String[] names = {SULFURAS, BRIE, PASS, CONJURED, "toto"};
        Integer[] prices = {0, 20};
        Integer[] qualities = {-1, 0, 1, 5, 6, 10, 50};
        CombinationApprovals.verifyAllCombinations(this::doTest, names, prices, qualities);
    }

    private Item doTest(String name, int sellIn, int quality) {
        //GIVEN
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        //WHEN
        app.updateQuality();
        return app.items[0];
    }

}
