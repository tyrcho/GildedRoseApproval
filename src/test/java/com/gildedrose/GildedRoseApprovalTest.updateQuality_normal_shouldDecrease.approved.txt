[Sulfuras, Hand of Ragnaros, 0, -1] => [0, -1] 
[Sulfuras, Hand of Ragnaros, 0, 0] => [0, 0] 
[Sulfuras, Hand of Ragnaros, 0, 50] => [0, 50] 
[Sulfuras, Hand of Ragnaros, 10, -1] => [10, -1] 
[Sulfuras, Hand of Ragnaros, 10, 0] => [10, 0] 
[Sulfuras, Hand of Ragnaros, 10, 50] => [10, 50] 
[Sulfuras, Hand of Ragnaros, 20, -1] => [20, -1] 
[Sulfuras, Hand of Ragnaros, 20, 0] => [20, 0] 
[Sulfuras, Hand of Ragnaros, 20, 50] => [20, 50] 
[Aged Brie, 0, -1] => [-1, 1] 
[Aged Brie, 0, 0] => [-1, 2] 
[Aged Brie, 0, 50] => [-1, 50] 
[Aged Brie, 10, -1] => [9, 0] 
[Aged Brie, 10, 0] => [9, 1] 
[Aged Brie, 10, 50] => [9, 50] 
[Aged Brie, 20, -1] => [19, 0] 
[Aged Brie, 20, 0] => [19, 1] 
[Aged Brie, 20, 50] => [19, 50] 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => [-1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => [-1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => [-1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 10, -1] => [9, 1] 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => [9, 2] 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => [9, 50] 
[Backstage passes to a TAFKAL80ETC concert, 20, -1] => [19, 0] 
[Backstage passes to a TAFKAL80ETC concert, 20, 0] => [19, 1] 
[Backstage passes to a TAFKAL80ETC concert, 20, 50] => [19, 50] 
[Conjured, 0, -1] => [-1, -1] 
[Conjured, 0, 0] => [-1, 0] 
[Conjured, 0, 50] => [-1, 46] 
[Conjured, 10, -1] => [9, -1] 
[Conjured, 10, 0] => [9, 0] 
[Conjured, 10, 50] => [9, 48] 
[Conjured, 20, -1] => [19, -1] 
[Conjured, 20, 0] => [19, 0] 
[Conjured, 20, 50] => [19, 48] 
[toto, 0, -1] => [-1, -1] 
[toto, 0, 0] => [-1, 0] 
[toto, 0, 50] => [-1, 48] 
[toto, 10, -1] => [9, -1] 
[toto, 10, 0] => [9, 0] 
[toto, 10, 50] => [9, 49] 
[toto, 20, -1] => [19, -1] 
[toto, 20, 0] => [19, 0] 
[toto, 20, 50] => [19, 49] 
